langchain4j-vertex-ai/src/main/java/dev/langchain4j/model/vertexai/internal/VertexAIDocument.java

```diff
@@ -0,0 +1,32 @@
+package dev.langchain4j.model.vertexai.internal;
+
+import com.google.gson.Gson;
+import dev.langchain4j.data.segment.TextSegment;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+/**
+ * This class is used to serialize the document to JSON.
+ */
+@Getter
+@NoArgsConstructor
+@AllArgsConstructor
+public class VertexAIDocument {
+    private static final Gson GSON = new Gson();
+
+    private String memberOfIndex;
+    private TextSegment segment;
```

Comment:
I would probably instead define `String text` and `Map<String, String> metadata` as fields in this class instead of `TextSegment` and map it manually because the structure of `TextSegment` might change at some point

Reply:
Done!